libandroid-navigation-ui/src/test/java/com/mapbox/services/android/navigation/ui/v5/route/MapRouteLineTest.java

```diff
@@ -207,4 +336,23 @@ private MapboxMap buildMockMap() {
     when(mapboxMap.getStyle()).thenReturn(style);
     return mapboxMap;
   }
+
+  private FeatureProcessingTask buildFeatureProcessingTask(List<DirectionsRoute> routes, MapRouteLine routeLine, Handler handler) {
+    OnRouteFeaturesProcessedCallback featuresProcessedCallback = routeLine.retrieveRouteFeaturesProcessedCallback();
```

Comment:
cannot we send our own callback to avoid use `countdownlatch`?

Reply:
"Problem" is that the callback is executed as part of the `completion` from `FeatureProcessingTask` which runs in a separate `Thread`. As mentioned above, we can implement the tasks as `Runnable`s and create the thread in the call-site. That way we'll be able to execute the `run` synchronously, avoiding `CountDownLatch`. We can 👀 later, as part of a follow up PR.